components/nodemanager-service/tests/project_update_test.go

```diff
@@ -0,0 +1,2231 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	iam_v2 "github.com/chef/automate/api/interservice/authz/v2"
+	"github.com/chef/automate/components/nodemanager-service/api/manager"
+	"github.com/chef/automate/components/nodemanager-service/api/nodes"
+	"github.com/chef/automate/components/nodemanager-service/config"
+	"github.com/chef/automate/components/nodemanager-service/pgdb"
+	"github.com/gofrs/uuid"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProjectUpdate(t *testing.T) {
+	ctx := context.Background()
+	db, err := createPGDB()
+	require.NoError(t, err)
+
+	timestamp, err := ptypes.TimestampProto(time.Now())
+	require.NoError(t, err)
+
+	cases := []struct {
+		description        string
+		projectsData       []*nodes.ProjectsData
+		projectRules       map[string]*iam_v2.ProjectRules
+		originalProjectIDs []string
+		expectedProjectIDs []string
+	}{
+		// Environment
+		{
+			description: "Environment - updating project",
```

Comment:
Would suggest make your test names more uniform in the format of
"`<characteristic> <expectation> <when what happens>`"
For this test you give the expectation but not the what, so suggest change to:
 "Environment - project updates with single rule, single matching condition"

Right now, you give the expectation explicitly just for the first 2 tests.
So test 3 should change from:
"Environment - Single rule two matching conditions"
to
"Environment - project updates with single rule, two matching conditions"

And test 4 from:
"Environment - Single rule, one non-matching condition"
to
"Environment - project dropped with single rule, one non-matching condition"